### DISCUSSION TITLE
NTP server from DHCP

### DISCUSSION THREAD

**Discussion by yearski** [arendst/Tasmota#11721] [Q&A] [answered]
In versions prior to 9.2, tasmota devices would use the NTP server provided by the DHCP info. @Jason2866 stated that the functionality was intentionally removed: https://github.com/arendst/Tasmota/issues/11132#issuecomment-786490071 but I'm not convinced. Was this done in the tasmota code? I've searched and can't find it (either before or after a change). Did it change in the ESP8266/Arduino library? Coincidentally, @arendst made this commit: [Add NTP fallback servers x.pool.ntp.org](https://github.com/arendst/Tasmota/commit/c783ce445bd0604e49abb5cc12b829c02f3577a0) which seems suspect but I tried compiling without this commit and still couldn't get NTP to provision from DHCP.

Would there be an easy way in the "fallback" logic to try the server provided by DHCP? If someone could give me a hint about how to access the DHCP response, I would try to submit a patch. Is it from the arduino library? Is the DHCP info saved beyond setting the host IP?

**Comment by sfromis** [comment]
arendst has made an unambiguous statement about having rewritten the NTP handling, not just done some workaround or deactivation. I've got no step-by-step explanation, but [this](https://github.com/arendst/Tasmota/commit/488a360d5b58b79cd450a249e0c26d459817d5d1) commit (https://github.com/arendst/Tasmota/issues/9739) does include a lot of NTP changing code.

**Comment by Jason2866** [comment, marked as answer]
The providing of the NTP servers via DHCP was removed because many users have routers which are not capable doing this and run into a issue. Maybe i mentioned the wrong commit where the chanve is done.

**Reply by yearski** [reply]
Was it in tasmota codebase? I've been looking at changes between 9.1 and 9.2 and can't find anything suspect except the NTP fallback stuff which I think I ruled out (assuming I correctly reverted those few lines and compiled). Hence my question if this really was intentional.

**Reply by sfromis** [reply]
Remember that Tasmota sits on top of Arduino Core containing lots of network-related code. Sometimes that code turns out to not work very well, and Tasmota works around that. Therefore it does not make sense to hunt by comparing Tasmota versions. The commit I linked has the new NTP sync code.
